Tests/CommunicationLayer/CommDeviceControl/UdpMessageForwarderTest.cpp

```
@@ -0,0 +1,203 @@
+
+    // TODO test that disconnecting the sender does not destroy messages in the queue
+
+    /**
+   * @brief SetUp will set up the receiver to verify messages are being sent, as well as mocking the settings to be used by the UdpMessageForwarder
+   */
+    virtual void SetUp()
+    {
+        settings_.reset(new MockSettings());
+        receiver = Channel::Create(MOCK_IP.toStdString(), MOCK_PORT);
+        receiver->DeclareQueue(MOCK_QUEUE.toStdString(), false, true, false, false);
+
+        receiver->BindQueue(MOCK_QUEUE.toStdString(), MOCK_EXCHANGE.toStdString(), MOCK_ROUTING_KEY.toStdString());
+
+        EXPECT_CALL(*settings_, ipAddress())
+                .WillRepeatedly(Return(MOCK_IP));
+        EXPECT_CALL(*settings_, queueName())
+                .WillRepeatedly(Return(MOCK_QUEUE));
+        EXPECT_CALL(*settings_, udpPort())
+                .WillRepeatedly(Return(MOCK_PORT));
+        EXPECT_CALL(*settings_, routingKey())
+                .WillRepeatedly(Return(MOCK_ROUTING_KEY));
+    }
+
+    virtual void TearDown() {
+        receiver->PurgeQueue(MOCK_QUEUE.toStdString());
+        receiver->DeleteQueue(MOCK_QUEUE.toStdString());
+        receiver.reset();
+    }
+
+    // Test Methods
+
+    void sendMessage(const QString& message, bool resetForwarder);
+    void receiveMessage(QString& actual, bool setupConsume);
+    /**
+     * @brief setup the receiver for usage
+     */
+    void setupReceiver();
+    /**
+     * @brief reset the receiver (clear and delete queue)
+     */
+    void cleanReceiver();
+};
+
+void UdpMessageForwarderTest::sendMessage(const QString& message, bool resetForwarder) {
+    if (resetForwarder)
+        forwarder.reset(new UdpMessageForwarder(*settings_));
+    QByteArray expectedBytes = QByteArray();
+    expectedBytes.append(message);
+    try {
+        forwarder->forwardData(expectedBytes);
+    } catch (quint16 err) {
+        qDebug() << "Failed to forward data with exception: " << err;
+    }
```

No need to have a try catch here, it's fine and less verbose to just let the forwarder throw an exception and crash the test.